cpp/src/datetime/datetime_ops.cu

```diff
@@ -290,6 +362,64 @@ std::unique_ptr<column> add_calendrical_months(column_view const& timestamp_colu
   return output;
 }
 
+std::unique_ptr<column> ceil_general(column_view const& column,
+                                     datetime_component component,
+                                     rmm::cuda_stream_view stream,
+                                     rmm::mr::device_memory_resource* mr)
+{
+  switch (component) {
+    case datetime_component::DAY:
+      return cudf::type_dispatcher(column.type(),
+                                   dispatch_ceil<detail::ceil_timestamp<datetime_component::DAY>>{},
```

Comment:
can we pass the duration type to dispach_ceil, to avoid another switch in there?